pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -487,15 +487,23 @@ func (e *EndpointController) syncService(key string) {
 	} else {
 		newEndpoints.Annotations[endpoints.PodHostnamesAnnotation] = serializedPodHostNames
 	}
-	if len(currentEndpoints.ResourceVersion) == 0 {
+
+	createEndpoints := len(currentEndpoints.ResourceVersion) == 0
+	if createEndpoints {
 		// No previous endpoints, create them
 		_, err = e.client.Endpoints(service.Namespace).Create(newEndpoints)
 	} else {
 		// Pre-existing
 		_, err = e.client.Endpoints(service.Namespace).Update(newEndpoints)
 	}
 	if err != nil {
-		glog.Errorf("Error updating endpoints: %v", err)
+		if createEndpoints && errors.IsForbidden(err) {
+			// if a namespace is terminating, endpoint creation is not allowed by default.
+			// rather than flood the logs with errors for what is normal system behavior, log at a lower level.
+			glog.V(5).Infof("Forbidden from creating endpoints: %v", err)
```

Comment:
Could potentially hide permission issues in the future if the endpoints controller doesn't have perm to create stuff.   But it's better than spamming the logs, which is very common.